settings/resource_default_namespace_settings.go

```diff
@@ -0,0 +1,178 @@
+func NewDefaultNamespaceSettingsAPI(ctx context.Context, m any) DefaultNamespaceSettingsAPI {
+	client := m.(*common.DatabricksClient)
+	return DefaultNamespaceSettingsAPI{client, ctx}
+}
+
+// DefaultNamespaceSettingsAPI exposes the Default Namespace Settings API
+type DefaultNamespaceSettingsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func (a DefaultNamespaceSettingsAPI) isEtagVersionError(err error) bool {
+	var aerr *apierr.APIError
+	if !errors.As(err, &aerr) {
+		return false
+	}
+	return aerr.StatusCode == http.StatusNotFound || (aerr.StatusCode == http.StatusConflict && aerr.ErrorCode == "RESOURCE_CONFLICT")
+}
+
+func (a DefaultNamespaceSettingsAPI) getEtagFromError(err error) (string, error) {
+	if !a.isEtagVersionError(err) {
+		return "", err
+	}
+	errorInfos := apierr.GetErrorInfo(err)
+	if len(errorInfos) > 0 {
+		metadata := errorInfos[0].Metadata
+		if etag, ok := metadata["etag"]; ok {
+			return etag, nil
+		}
+	}
+	return "", fmt.Errorf("error fetching the default workspace namespace settings: %w", err)
+}
+
+func (a DefaultNamespaceSettingsAPI) DeleteWithRetry(etag string) (string, error) {
+	etag, err := a.executeDelete(etag)
+	if err == nil {
+		return etag, nil
+	}
+	etag, err = a.getEtagFromError(err)
+	if err != nil {
+		return "", err
+	}
+	return a.executeDelete(etag)
+}
+
+func (a DefaultNamespaceSettingsAPI) executeDelete(etag string) (string, error) {
+	var response settings.DefaultNamespaceSetting
+	err := a.client.DeleteWithResponse(a.context, "/settings/types/default_namespace_ws/names/default", map[string]string{
+		"etag": etag,
+	}, &response)
+	if err != nil {
+		return "", err
+	}
+	return response.Etag, nil
+}
+
+func (a DefaultNamespaceSettingsAPI) UpdateWithRetry(request settings.UpdateDefaultWorkspaceNamespaceRequest) (string, error) {
+	etag, err := a.executeUpdate(request)
+	if err == nil {
+		return etag, nil
+	}
+	etag, err = a.getEtagFromError(err)
+	if err != nil {
+		return "", err
+	}
+	request.Setting.Etag = etag
+	return a.executeUpdate(request)
+}
+
+func (a DefaultNamespaceSettingsAPI) executeUpdate(request settings.UpdateDefaultWorkspaceNamespaceRequest) (string, error) {
+	var response settings.DefaultNamespaceSetting
+	err := a.client.PatchWithResponse(a.context, "/settings/types/default_namespace_ws/names/default", request, &response)
+	if err != nil {
+		return "", err
+	}
+	return response.Etag, nil
+}
+
+func (a DefaultNamespaceSettingsAPI) Read(etag string) (settings.DefaultNamespaceSetting, error) {
+	var setting settings.DefaultNamespaceSetting
+	err := a.client.Get(a.context, "/settings/types/default_namespace_ws/names/default", map[string]string{
+		"etag": etag,
+	}, &setting)
+	return setting, err
+}
+
+var resourceSchema = common.StructToSchema(settings.DefaultNamespaceSetting{},
+	func(s map[string]*schema.Schema) map[string]*schema.Schema {
+		delete(s, "etag")
```

Comment:
@hectorcast-db Question about these fields: I stumbled upon https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#computed, it looks like `computed` is used to indicate resource fields that are not user configurable. For all settings, etag can not be created by user - user can only pass in an etag returned from the API. For this setting, the setting name is also not user configurable since it's a singleton setting.

What's the reason of using delete here? Shouldn't we use the computed keyword?